File: files/300-399/398/CH-398.R
library(tidyverse)
library(readxl)

path <- "300-399/398/CH-398  Replacement.xlsx"
input <- read_excel(path, range = "B3:E10")
test <- read_excel(path, range = "G3:J10")

result = input |>
  mutate(`Product ID` = lead(`Product ID`, default = first(`Product ID`)))

all.equal(result$`Product ID`, test$`Product ID`)
# [1] TRUE

File: files/300-399/398/CH-398.py
import pandas as pd

path = "300-399/398/CH-398  Replacement.xlsx"
input = pd.read_excel(path, usecols="B:E", nrows=8, skiprows=2)
test = pd.read_excel(path, usecols="G:J", nrows=8, skiprows=2)

result = input.copy()
result["Product ID"] = result["Product ID"].shift(-1, fill_value=result["Product ID"].iloc[0])

print(result["Product ID"].equals(test["Product ID.1"]))
# Output: True